keras/utils/data_utils.py

```diff
@@ -55,23 +56,169 @@ def chunk_read(response, chunk_size=8192, reporthook=None):
     from six.moves.urllib.request import urlretrieve
 
 
+class Archive(object):
+    """ Defines an inteface to extract archive files like tar and zip files.
+    """
+    signature = None
+    offset = 0
+    file_type = None
+    mime_type = None
+    file_path = None
+
+    def __init__(self, file_path):
+        self.file_path = file_path
+
+    @classmethod
+    def is_match(cls, fpath):
+        """ Returns True if the file's signature matches the supported archive format.
+        """
+        match = False
+        if cls.signature is None:
+            return match
+        with open(fpath, 'rb') as f:
+            f.seek(cls.offset)
+            match = f.read(len(cls.signature)).startswith(cls.signature)
+        return match
+
+    def open(self, mode='rb'):
+        """ Open the file, or archive object in subclasses.
+        """
+        return open(self.file_path, mode)
+
+    def extractall(self, archive, path="."):
+        """ Extract all archive contents if available.
+        """
+        return None
+
+    def extractall_if_match(self, path="."):
+        if (self.file_type is None) or not self.is_match(self.file_path):
+            return False
+        with self.open() as archive:
+            try:
+                self.extractall(archive, path=path)
+            except (Exception, KeyboardInterrupt) as e:
+                if os.path.exists(path):
+                    if os.path.isfile(path):
+                        os.remove(path)
+                    else:
+                        shutil.rmtree(path)
+                raise
+        return True
+
+
+class TarArchive(Archive):
```

Comment:
I had started on two additional compression formats straight bz2 and lzma could be added also. 

Fair enough I'll look into just using a few if statements in extract_archive